model/minsky.cc

```diff
@@ -264,6 +265,35 @@ namespace minsky
     GroupPtr g(new Group);
     canvas.setItemFocus(model->addGroup(g));
     m.populateGroup(*g);
+    g->resizeOnContents();	  
+  }
+
+    // Fix for ticket 1080
+  void Minsky::pasteItems()
+  {
+    istringstream is(getClipboard());
+    xml_unpack_t unpacker(is);
+    schema2::Minsky m;
+    xml_unpack(unpacker, "Minsky", m);
+    GroupPtr g(new Group);
+    canvas.setItemFocus(model->addGroup(g));
+    m.populateGroup(*g);
+    if (auto p=g->group.lock())
+    {
+	   p->moveContents(*g); 
+	   canvas.item.reset();
+       model->recursiveDo(&GroupItems::items,
```

Comment:
No need to do a recursiveDo. It suffices to loop over items and groups of g, and call insertItem and insertGroup on that. I don't think there is any scenario where any items could be duplicated.

Reply:
Okay, I've done this, but I don't know if it is correct. Test t0036a.sh fails with this new change:

  void Minsky::pasteItems()
  {
    istringstream is(getClipboard());
    xml_unpack_t unpacker(is);
    schema2::Minsky m;
    xml_unpack(unpacker, "Minsky", m);
    GroupPtr g(new Group);
    canvas.setItemFocus(model->addGroup(g));
    m.populateGroup(*g);
    **auto p = model;
    p->moveContents(*g);
    for (auto& i: p->items)
      if (i->visible()) {
        canvas.selection.ensureItemInserted(i);
	}
    for (auto& i: p->groups)
      if (i->visible()) {
        canvas.selection.ensureGroupInserted(i);
	}**
    g->resizeOnContents();
  }

Reply:
Here is the latest solution (insideminsky::paste()) I have committed:

    // Fix for ticket 1080    
    auto p = model;
    p->moveContents(*g);
    for (auto& i: p->items) {   
        canvas.selection.ensureItemInserted(i);  
        canvas.setItemFocus(model->addItem(i)); 
	}
    g->resizeOnContents();   
    model->removeGroup(*g);